packages/opentelemetry-tracing/src/BasicTracerFactory.ts

```diff
@@ -0,0 +1,54 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AbstractTracerFactory } from '@opentelemetry/core';
+import { BasicTracer } from './BasicTracer';
+import { BasicTracerConfig } from './types';
+import { SpanProcessor } from './SpanProcessor';
+
+export class BasicTracerFactory extends AbstractTracerFactory<BasicTracer> {
+  private _config?: BasicTracerConfig;
+  private _spanProcessors: SpanProcessor[] = [];
+
+  constructor(config?: BasicTracerConfig) {
+    super();
+    this._config = config;
+  }
+
+  /**
+   * addSpanProcessor adds a {@link SpanProcessor} to the factory, applying it
+   * to all new and existing tracers.
+   */
+  addSpanProcessor(spanProcessor: SpanProcessor): void {
```

Comment:
Why is this not on `AbstractTracerFactory`?

Reply:
API vs SDK. Again, addSpanProcessor is a sdk concept and doesn't exist at the API level. If this is going to be in core, it's should be at the level of the API.